src/rules/terraform.ts

```diff
@@ -0,0 +1,87 @@
+import JMESPath = require('jmespath');
+import Handlebars = require('handlebars');
+import {promises as fsPromises} from 'fs';
+
+import {BaseRule, Evaluation, isAndEvaluation, RuleType} from '.';
+import {ScanResult} from '../scanner';
+
+export interface TerraformTarget {
+  type: RuleType.Terraform;
+  plan: TerraformResource[];
+}
+
+export interface TerraformResource {
+  address: string;
+}
+
+export class TerraformRule implements BaseRule<TerraformTarget> {
+  type: RuleType.Terraform;
+  name: string;
+  description: string;
+  evaluation: Evaluation;
+  recommendation: string;
+
+  constructor(rule: {
+    type: RuleType.Terraform;
+    name: string;
+    description: string;
+    evaluation: Evaluation;
+    recommendation: string;
+  }) {
+    this.type = rule.type;
+    this.name = rule.name;
+    this.description = rule.description;
+    this.evaluation = rule.evaluation;
+    this.recommendation = rule.recommendation;
+  }
+
+  async execute(target: TerraformTarget): Promise<ScanResult> {
+    const results = this.evaluate(this.evaluation, target.plan);
+    return Promise.resolve(this.toScanResult(results));
+  }
+
+  static async getTarget(planFilePath: string): Promise<TerraformTarget> {
+    const data = await fsPromises.readFile(planFilePath, 'utf8');
+    const plan = JSON.parse(data);
+    return {
+      type: RuleType.Terraform,
+      plan,
```

Comment:
I think this is the only bit that's fuzzy for me - at least as far as the types are defined. I think plan will come back as an object with

```js
{
  format_version: "",
  terraform_version: "",
  planned_values: {},
  //and so on
}
```

But the type of `TerraformTarget` seems to indicate that plan is an array of objects with an `address` property